File: 0896-MonotonicArray/MonotonicArray.go
package monotonicarray

func isMonotonic(nums []int) bool {
	// notassigned - 0, increasing -> 1, decreasing 2
	status := 0

	prev := nums[0]
	for i := 1; i < len(nums); i++ {
		if prev < nums[i] {
			if status == 2 {
				return false
			}
			status = 1
		}
		if prev > nums[i] {
			if status == 1 {
				return false
			}
			status = 2
		}

		prev = nums[i]
	}

	return true
}

File: 0896-MonotonicArray/MonotonicArray.java

class MonotonicArray {
    class Solution {
        public boolean isMonotonic(int[] nums) {
            Boolean isIncreasing = null;
            if(nums.length == 1) return true;
            
            int prev = nums[0];
            for(int i = 1; i < nums.length; i++){
                if(nums[i] > prev){
                    if(isIncreasing != null && isIncreasing == false) return false;
                    isIncreasing = true;
                }
                if(nums[i] < prev){
                    if(isIncreasing != null && isIncreasing == true) return false;
                    isIncreasing = false;
                }
                prev = nums[i];
            }
            return true;
        }
    }
}
